lib/Elastica/Reindex.php

```diff
@@ -32,84 +46,94 @@ class Reindex
         $reindexEndpoint->setBody($body);
 
-        $this->_oldIndex->getClient()->requestEndpoint($reindexEndpoint);
-        $this->_newIndex->refresh();
+        $this->_lastResponse = $this->_oldIndex->getClient()->requestEndpoint($reindexEndpoint);
 
-        return $this->_newIndex;
+        return $this->_lastResponse;
     }
 
-    protected function _getBody($oldIndex, $newIndex, $options)
+    protected function _getBody($oldIndex, $newIndex, $params)
     {
         $body = \array_merge([
-            'source' => $this->_getSourcePartBody($oldIndex, $options),
-            'dest' => $this->_getDestPartBody($newIndex, $options),
-        ], $this->_resolveBodyOptions($options));
+            'source' => $this->_getSourcePartBody($oldIndex, $params),
+            'dest' => $this->_getDestPartBody($newIndex, $params),
+        ], $this->_resolveBodyOptions($params));
+
+        $body = $this->_setBodyScript($body);
 
         return $body;
     }
 
-    protected function _getSourcePartBody(Index $index, array $options)
+    protected function _getSourcePartBody(Index $index, array $params)
     {
         $sourceBody = \array_merge([
             'index' => $index->getName(),
-        ], $this->_resolveSourceOptions($options));
+        ], $this->_resolveSourceOptions($params));
 
         $sourceBody = $this->_setSourceQuery($sourceBody);
         $sourceBody = $this->_setSourceType($sourceBody);
 
         return $sourceBody;
     }
 
-    protected function _getDestPartBody(Index $index, array $options)
+    protected function _getDestPartBody(Index $index, array $params)
     {
-        return \array_merge([
+        $destBody = \array_merge([
             'index' => $index->getName(),
-        ], $this->_resolveDestOptions($options));
+        ], $this->_resolveDestOptions($params));
+
+        return $destBody;
     }
 
-    private function _resolveSourceOptions(array $options)
+    private function _resolveSourceOptions(array $params)
     {
-        return \array_intersect_key($options, [
+        return \array_intersect_key($params, [
             self::TYPE => null,
             self::QUERY => null,
+            self::SORT => null,
+            self::SOURCE => null,
+            self::REMOTE => null,
+            self::SLICE => null,
         ]);
     }
 
-    private function _resolveDestOptions(array $options)
+    private function _resolveDestOptions(array $params)
     {
-        return \array_intersect_key($options, [
+        return \array_intersect_key($params, [
             self::VERSION_TYPE => null,
             self::OPERATION_TYPE => null,
         ]);
     }
 
-    private function _resolveBodyOptions(array $options)
+    private function _resolveBodyOptions(array $params)
     {
-        return \array_intersect_key($options, [
+        return \array_intersect_key($params, [
             self::SIZE => null,
             self::CONFLICTS => null,
         ]);
     }
 
-    /**
```

Comment:
Why did you remove these comments?

Reply:
To make the code more homogeneous and less noisy - lots of methods are self-documented via type-hint, this PHP DocBlock don't provide any useful information and is on a private method.

Should I put it back?

Reply:
How will doc blocks generated by phpdoc look like without the docs? Will an IDE know the type that is returned? If the behaviour with or without stays the same, happy to remove it.

Reply:
Those are private methods so I don't think it changes anything for the end user.

Also behavior is really good on IDE like phpStorm too. I'm even going one step further by adding return type hint as Elastica is 7.0 :+1: